src/lib/containers/oauth/CreateOAuthClient.tsx

```diff
@@ -0,0 +1,280 @@
+        const oAuthClient: OAuthClient = {
+          client_id: client?.client_id,
+          client_name: clientName,
+          redirect_uris: redirectUrisArray ?? [''],
+          policy_uri: policyUri,
+          client_uri: clientUri,
+          sector_identifier_uri: sectorUri ?? '',
+          tos_uri: tosUri,
+          etag: client?.etag,
+        }
+        setUpdatedClient(oAuthClient)
+        if (warnTrigger === true) {
+          setIsShowingConfirmModal(true)
+          hide()
+        } else {
+          if (isEdit) {
+            mutate({ action: 'UPDATE', client: oAuthClient })
+          } else {
+            mutate({ action: 'CREATE', client: oAuthClient })
+          }
+        }
+      }
+    } catch (err) {
+      console.log(err.reason)
+    }
+  }
+
+  return (
+    <div>
+      <Modal
+        show={isShowingModal}
+        animation={false}
+        backdrop="static"
+        onHide={hide}
+        size="lg"
+        className="OAuthDialog"
+      >
+        <Modal.Header closeButton>
+          <Modal.Title>
+            <Typography variant="headline1">
+              {isEdit ? 'Client Details' : 'Create New OAuth Client'}
+            </Typography>
+          </Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+          <Typography variant="body1">
+            To protect you and your users, your consent screen and application
+            will need to be verified by Sage Bionetworks. Before your consent
+            screen and application are verified by Sage Bionetworks, you can
+            still test your application with limitations. Learn more about how
+            your app will behave before it's verified.
+          </Typography>
+
+          {isEdit && (
+            <Typography style={{ marginTop: '16px' }} variant="label">
+              Client ID: {client?.client_id}
+            </Typography>
+          )}
+          <Row>
+            <Col lg={6} md={6} sm={12} xs={12}>
+              <Form.Group className="required">
+                <Form.Label>Client Name</Form.Label>
+                <Form.Control
+                  required
+                  onChange={e => setClientName(e.target.value)}
+                  placeholder="Client Name"
+                  type="text"
+                  value={clientName}
+                />
+              </Form.Group>
+            </Col>
+            <Col lg={6} md={6} sm={12} xs={12}>
+              <Form.Label className="required">Client Homepage</Form.Label>
+              <Form.Control
+                onChange={e => setClientUri(e.target.value)}
+                placeholder="https://"
+                type="text"
+                value={clientUri}
+              />
+            </Col>
+          </Row>
+          <Row>
+            {!isEdit && (
+              <>
+                <Col lg={6} md={6} sm={12} xs={12}>
+                  <Form.Group className="required">
+                    <Form.Label>Redirect URI(s)</Form.Label>
```

Comment:
May be worth adding a tooltip or HelpPopover to indicate that the URIs should be comma separated.

I'm not 100% sure comma-separated is the right control--Is it possible that a URI contains a comma?

Reply:
It is technically possible for a URI to contain a comma but is considered bad practice. Do you think we should still go with comma separation?